src/lexer.coffee

```diff
@@ -416,13 +416,13 @@ exports.Lexer = class Lexer
         @token 'REGEX_START', '(',    {length: 0, origin}
         @token 'IDENTIFIER', 'RegExp', length: 0
         @token 'CALL_START', '(',      length: 0
-        @mergeInterpolationTokens tokens, {double: yes, heregex: {flags}, endOffset: end}, (str) =>
+        @mergeInterpolationTokens tokens, {double: yes, heregex: {flags}, endOffset: end - flags.length, quote: '///'}, (str) =>
           @validateUnicodeCodePointEscapes str, {delimiter}
         if flags
           @token ',', ',',                    offset: index - 1, length: 0
           @token 'STRING', '"' + flags + '"', offset: index - 1, length: flags.length
-        @token ')', ')',                      offset: end - 1,   length: 0
-        @token 'REGEX_END', ')',              offset: end - 1,   length: 0
+        @token ')', ')',                      offset: end,       length: 0
```

Comment:
Without this change the location data for heregexes was one character too short

This change makes sense, the generated tokens denoting the end of the heregex should be zero-length tokens between last character of the heregex and the following character